### DISCUSSION TITLE
Varying text color by value?

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#1085] [Q&A]
I'd like to show a bunch (around 100-200) of numbers as separate `Text` objects, where the color of the text is based on the value shown. As it seems `TextStyle` is the only way to set the color of a piece of `Text` I'm wondering what's worse, creating a new `TextStyle` each time a value changes (potentially each frame), or giving each `Text` object its own `TextStyle` and then updating just the color in the `Material`? I suspect the latter is the best approach?

**Comment by maluoi** [comment]
There is actually a couple of `Text.Add` overloads that take a color as a parameter! I'd use a white `TextStyle`, and then use those overloads.
